### ISSUE TITLE
Unexpected file found in package directory lwc\jsconfig.json

### ISSUE THREAD

**Issue by bard-casi** [forcedotcom/cli#1349]
### Summary

On 12/7/2021, our lead made us aware of an issue in building our sandbox, where they received the following error: 

> Deployment failed, with error message: Unexpected file found in package directory: /home/vsts/work/1/s/src/smart/main/default/lwc/jsconfig.json. 

However, they were able to build a scratch on their end without issue, so this problem was initially ignored. 

Then on 12/13/2021, I attempted to run `sfdx force:source:push -f` to a scratch org myself. Since then I have been getting the 'Unexpected file found in package directory lwc\jsconfig.json' error every single time I attempt to use push my source to a scratch org.

### Steps To Reproduce:

1.  Run `sfdx force:org:create -f` to create scratch org
2.  Deploy metadata by running `sfdx force:mdapi:deploy -f`
3.  Run `sfdx force:source:push -f` to attempt to push source to default scratch org

### Expected result

Source is successfully pushed to the default scratch org.

### Actual result

The following error occurs: 

> ERROR running force:source:push:  Unexpected file found in package directory: C:\Salesforce IT Repo\Salesforce\src\smart\main\default\lwc\jsconfig.json

### Additional information

- We have the lwc\jsconfig.json files in our .forceignore:
![image](https://user-images.githubusercontent.com/97206371/148322299-9de44d72-25df-4efe-8290-e3a22cef69bd.png)

- Here is a screenshot of the error:
![image](https://user-images.githubusercontent.com/97206371/148321307-f71e4c87-0f58-4492-9088-b5f9b0e6a5cd.png)

- I cannot point to anything in particular that could have caused this. We have scoured our commits since November, and no changes made on our end should have lead to this suddenly beginning to occur.

- My colleagues and I have tried many, many different ways to try and get around this issue, including:

1. Changing CLI version: CLI 7.129.x (npm), CLI 7.130.x (npm), CLI 7.131.x (npm), CLI 96.1.x (npm) and a Non-npm install: CLI 7.132.0
2. Run scratch build against an older branch of ours (before this error began occurring)
3. Delete the jsconfig.json file, which leads to a slew of other errors
4. Cloning Salesforce project into new directory
5. Clearing all temp files, appdata, etc related to SFDX and npm
6. Uninstalling and deleting EVERYTHING locally relating to Salesforce, npm, SFDX, and VSCode
7. And every combination of the above you can think of

The ONLY thing that we have found to get around this is to use the command `sfdx force:source:beta:push`, which DOES successfully push the data to the scratch org. However, this causes a lot of other issues with git AND I am unable to properly go through with our dev process because I run into errors when attempting to push to and pull from the default scratch org once changes have been made. So this is not a valid fix for our problem.

**Salesforce Extension Version in VS Code**: Salesforce Extension Pack v53.12.0

**SFDX CLI Version**: sfdx-cli/7.132.0

**OS and version**: Windows 10 Enterprise

**Comment by maggiben** [comment]
Hi, as a workaround can you try using `.forceignore` on that file ? like explained [here[(https://developer.salesforce.com/docs/atlas.en-us.sfdx_dev.meta/sfdx_dev/sfdx_dev_exclude_source.htm) 
Also what was the latest version of the CLI that did not produce this problem ?

**Comment by bard-casi** [comment]
I tried to explicitly add the entire file within the `.forceignore` rather than `**/jsconfig.json` which we already had in the .forceignore file, and got the same error:

> Deployment failed, with error message: Unexpected file found in package directory: src\smart\main\default\lwc\jsconfig.json

Actually, it seems like the entirety of the `.forceignore` file is not being recognized. When I use the command `sfdx force:source:ignored:list`, not a single ignored file is found despite us having many listed in the `.forceignore`. 
![image](https://user-images.githubusercontent.com/97206371/148464419-a5005a31-acec-449f-83b9-2a7b6311c47d.png)

We have not moved nor updated the `.forceignore` within the last few months.

The latest CLI that I used without issue was CLI 7.127.0. I went from CLI 7.127.0 to 7.130.01 and began to have issues shortly afterwards.

**Comment by bard-casi** [comment]
Figured this issue out (I believe). I can't pin down the reason that the `jsconfig.json` wasn't being ignored, as the syntax for that seemed to be proper. Nor can I figure out why this all of a sudden stopped working in general, but after replacing every \ (backward slash) in our `.forceignore` with a / (forward slash), it began to work.

Now getting a different set of errors, but at least we made it through one. :) Thanks for the help in looking into this, and apologies for any time it took to look into!
